complete-slice/complete-milestone blocked by planning-dispatch tools policy — cannot run verification bash commands

## Problem
`complete-slice` and `complete-milestone` units are HARD BLOCKED when they attempt to run verification bash commands (e.g. `make test`, `go vet`, `go test`) because the `unit-context-manifest.js` assigns them the `TOOLS_PLANNING_DISPATCH_REVIEW` policy. This policy restricts writes to `.gsd/` and blocks arbitrary bash execution.

The agent receives:
```
HARD BLOCK: unit "complete-slice" runs under tools-policy "planning-dispatch" 
— cannot edit "cmd/retry_integration_chaos_test.go" — writes are restricted to .gsd/.
```

This causes the slice/milestone closeout to fail artifact verification, which then triggers retry loops or stuck-loop detection.

## Root Cause
In `unit-context-manifest.js`, the manifests for `complete-slice` and `complete-milestone` declare:

```javascript
// unit-context-manifest.js:192-211 (validate-milestone)
"validate-milestone": {
    ...
    tools: TOOLS_PLANNING_DISPATCH_REVIEW,  // mode: "planning-dispatch"
    ...
}

// unit-context-manifest.js:299 (complete-slice)
"complete-slice": {
    ...
    tools: TOOLS_PLANNING_DISPATCH_REVIEW,  // mode: "planning-dispatch"
    ...
}

// unit-context-manifest.js:211 (complete-milestone)
"complete-milestone": {
    ...
    tools: TOOLS_PLANNING_DISPATCH_REVIEW,  // mode: "planning-dispatch"
    ...
}
```

`TOOLS_PLANNING_DISPATCH_REVIEW` is defined as:
```javascript
// unit-context-manifest.js:84-87
const TOOLS_PLANNING_DISPATCH_REVIEW = {
    mode: "planning-dispatch",
    allowedSubagents: ["reviewer", "security", "tester"],
};
```

The write-gate (`bootstrap/write-gate.js`) enforces that `planning-dispatch` mode blocks:
- Any file edit outside `.gsd/`
- Any `bash` command that is not read-only (grep, cat, git log, etc.)

But the **prompts** for `complete-slice` and `complete-milestone` instruct the agent to run verification commands and potentially fix issues. When the agent obeys the prompt, it hits the HARD BLOCK.

## Expected Behavior
Closeout units (`complete-slice`, `complete-milestone`, `validate-milestone`) that need to run verification bash commands and edit source files to fix issues must use a tools policy that permits those actions.

Two possible fixes:
1. **Change the manifest tools mode** for these units from `TOOLS_PLANNING_DISPATCH_REVIEW` to `TOOLS_ALL` (or a new `TOOLS_CLOSEOUT` that allows bash + edits but still restricts dangerous operations).
2. **Keep the manifest but change the prompt** to explicitly tell the agent NOT to run bash verification or edit source files during closeout, delegating all verification to subagents only.

Given that the current prompt already asks for verification commands, option 1 is safer — changing the manifest to match the prompt intent.

## Environment
- GSD version: 2.81.0
- Units affected: `complete-slice/M012/S05`, `complete-milestone/M009`, `validate-milestone/M012`

## Forensic Evidence
From the activity log for `complete-slice/M012/S05`:
```
HARD BLOCK: unit "complete-slice" runs under tools-policy "planning-dispatch" 
— cannot edit "cmd/retry_integration_chaos_test.go" — writes are restricted to .gsd/.
```

From `execute-task/M012/S05/T04`:
```
# command-line-arguments
vet: cmd/retry_integration_chaos_test.go:87:2: undefined: writeRuntimeConfig
Command exited with code 1
go test -short -count=1 ./...
```

This shows the project had broken code that needed fixing, but the closeout unit was blocked from editing it.

--- a/src/resources/extensions/gsd/tests/unit-context-manifest.test.ts
+++ b/src/resources/extensions/gsd/tests/unit-context-manifest.test.ts
@@ -230,15 +230,21 @@ test("#4934: tools.mode is one of the declared policies", () => {
   }
 });
 
-test('#4934: only execution units and complete-milestone may use tools.mode "all"', () => {
-  const allowedAllUnits = new Set(["execute-task", "reactive-execute", "complete-milestone"]);
+test('#4934: only execution units and closeout units may use tools.mode "all"', () => {
+  const allowedAllUnits = new Set([
+    "execute-task",
+    "reactive-execute",
+    "validate-milestone",
+    "complete-milestone",
+    "complete-slice",
+  ]);
   for (const [unitType, manifest] of Object.entries(UNIT_MANIFESTS)) {
     const mode = (manifest as { tools: { mode: string } }).tools.mode;
     if (mode === "all") {
       assert.ok(
         allowedAllUnits.has(unitType),
         `manifest "${unitType}" declares tools.mode = "all" but is not explicitly allowed. ` +
-        'Only execute-task, reactive-execute, and complete-milestone should have full source write access; ' +
+        'Only execute-task/reactive-execute and closeout units should have full source write access; ' +
         'planning/discuss/research units must use "planning" or "planning-dispatch" (or "docs" for rewrite-docs).',
       );
     }
@@ -268,6 +274,25 @@ test("#5453: complete-milestone uses all tools so bash verification is not plann
       result.block,
       false,
       `shouldBlockPlanningUnit must not block ${cmd} for complete-milestone: ${result.reason}`,
+    );
+  }
+});
+
+test("#5731: validate-milestone and complete-slice use all tools so closeout verification is not planning-dispatch blocked", () => {
+  for (const unitType of ["validate-milestone", "complete-slice"] as const) {
+    const manifest = UNIT_MANIFESTS[unitType];
+    assert.strictEqual(manifest.tools.mode, "all");
+    const result = shouldBlockPlanningUnit(
+      "bash",
+      "go test -short -count=1 ./...",
+      process.cwd(),
+      unitType,
+      manifest.tools,
+    );
+    assert.strictEqual(
+      result.block,
+      false,
+      `${unitType} must allow verification bash commands: ${result.reason}`,
     );
   }
 });
@@ -306,13 +331,11 @@ test('planning-dispatch mode is reserved for slice-level decomposition and compl
     "plan-slice",
     "research-slice",
     "refine-slice",
-    "complete-slice",
     "gate-evaluate",
     // Deep planning mode: research-project orchestrates 4 parallel research
     // subagents (stack/features/architecture/pitfalls). Subagent dispatch is
     // the unit's core mechanism — without it, the unit cannot do its job.
     "research-project",
-    "validate-milestone",
   ]);
   for (const [unitType, manifest] of Object.entries(UNIT_MANIFESTS)) {
     const mode = (manifest as { tools: { mode: string } }).tools.mode;

--- a/src/resources/extensions/gsd/unit-context-manifest.ts
+++ b/src/resources/extensions/gsd/unit-context-manifest.ts
@@ -409,11 +409,9 @@ export const UNIT_MANIFESTS: Record<UnitType, UnitContextManifest> = {
     codebaseMap: false,
     preferences: "active-only",
     contextMode: "verification",
-    // planning-dispatch: validation is a verification-fan-out unit. It reads
-    // the milestone surface and dispatches reviewer/security/tester subagents
-    // to report findings without touching user source. Write isolation to
-    // .gsd/ is preserved.
-    tools: TOOLS_PLANNING_DISPATCH_REVIEW,
+    // Validation may need to run shell verification commands and apply source
+    // fixes before milestone closeout can proceed.
+    tools: TOOLS_ALL,
     artifacts: {
       inline: ["roadmap", "slice-summary", "slice-uat", "requirements", "decisions", "templates"],
       excerpt: [],
@@ -518,10 +516,9 @@ export const UNIT_MANIFESTS: Record<UnitType, UnitContextManifest> = {
     codebaseMap: false,
     preferences: "active-only",
     contextMode: "verification",
-    // See complete-milestone — same rationale: dispatch to reviewer / security /
-    // tester subagents to fan out review work without bloating this unit's
-    // context.
-    tools: TOOLS_PLANNING_DISPATCH_REVIEW,
+    // Slice closeout may need to run shell verification commands and apply
+    // source fixes before completion can be recorded.
+    tools: TOOLS_ALL,
     artifacts: {
       // Phase 3 migration (#4782): matches today's actual
       // buildCompleteSlicePrompt inlining order. Overrides prepend +

